discovery-infra/tests/base_test.py

```diff
@@ -280,3 +281,7 @@ def check_status():
             sleep_seconds=90,
             waiting_for="controller to be running",
         )
+
+    @pytest.fixture()
+    def cluster_deployment_context(self):
```

Comment:
I think `cluster_deployment` is good for a fixture

Reply:
I agree, but on the other hand, I think it is more indicative to tell that this fixture is context so you'll know to use it as context manager, what do you think?

Reply:
Not critical. :)